.github/workflows/rust.yml

```diff
@@ -24,17 +24,6 @@ jobs:
         with:
           toolchain: ${{ matrix.rust }}
           override: true
-      - name: Fixes for MSRV
-        if: matrix.rust == '1.57.0'
-        run: |
-          cargo update
-          cargo update -p log --precise 0.4.18
-          cargo update -p tempfile --precise 3.6.0
-          cargo update -p flate2 --precise 1.0.26
-          cargo update -p minreq --precise 2.8.0
-          cargo update -p rustls --precise 0.20.8
-          cargo update -p webpki --precise 0.22.2
-          cargo update -p byteorder --precise 1.4.3
```

Comment:
We probably need to include something like this for 1.63.0. It's failing CI because deps aren't pinned afaict

Reply:
Yea the weird part is that when i set 1.63 for me locally it does compile
Ill investigate this

Reply:
thats how I run it, and it seems to work?
```
[I] ecode@pop-os: rust-payjoin [1.63-MSRV] » git status                                                                             [15:15:52]
On branch 1.63-MSRV
Your branch is up to date with 'origin/1.63-MSRV'.

nothing to commit, working tree clean
[I] ecode@pop-os: rust-payjoin [1.63-MSRV] » rustup show                                                                            [15:15:53]
Default host: x86_64-unknown-linux-gnu
rustup home:  /home/ecode/.rustup

installed toolchains
--------------------

stable-x86_64-unknown-linux-gnu
nightly-x86_64-unknown-linux-gnu
1.48-x86_64-unknown-linux-gnu
1.69-x86_64-unknown-linux-gnu (default)
1.70-x86_64-unknown-linux-gnu
1.71-x86_64-unknown-linux-gnu

active toolchain
----------------

1.69-x86_64-unknown-linux-gnu (default)
rustc 1.69.0 (84c898d65 2023-04-16)

[I] ecode@pop-os: rust-payjoin [1.63-MSRV] » rustup default 1.63                                                                    [15:15:56]
info: syncing channel updates for '1.63-x86_64-unknown-linux-gnu'
info: latest update on 2022-08-11, rust version 1.63.0 (4b91a6ea7 2022-08-08)
info: downloading component 'cargo'
info: downloading component 'clippy'
info: downloading component 'rust-docs'
info: downloading component 'rust-std'
info: downloading component 'rustc'
 54.3 MiB /  54.3 MiB (100 %)  42.2 MiB/s in  1s ETA:  0s
info: downloading component 'rustfmt'
info: installing component 'cargo'
info: installing component 'clippy'
info: installing component 'rust-docs'
 18.3 MiB /  18.3 MiB (100 %)  17.1 MiB/s in  1s ETA:  0s
info: installing component 'rust-std'
 26.1 MiB /  26.1 MiB (100 %)  19.8 MiB/s in  1s ETA:  0s
info: installing component 'rustc'
 54.3 MiB /  54.3 MiB (100 %)  23.0 MiB/s in  2s ETA:  0s
info: installing component 'rustfmt'
info: default toolchain set to '1.63-x86_64-unknown-linux-gnu'

  1.63-x86_64-unknown-linux-gnu installed - rustc 1.63.0 (4b91a6ea7 2022-08-08)

[I] ecode@pop-os: rust-payjoin [1.63-MSRV] » cargo b                                                                                [15:16:20]
error: could not find `Cargo.toml` in `/home/ecode/satoshi/rust-payjoin` or any parent directory
[I] ecode@pop-os: rust-payjoin [1.63-MSRV] » cd payjoin                                                                         101 [15:16:23]
[N] ecode@pop-os: payjoin [1.63-MSRV] » cargo b                                                                                     [15:16:29]
   Compiling libc v0.2.147
   Compiling bitcoin-private v0.1.0
   Compiling tinyvec_macros v0.1.1
   Compiling hex_lit v0.1.1
   Compiling bitcoin v0.30.1
   Compiling bech32 v0.9.1
   Compiling unicode-bidi v0.3.13
   Compiling percent-encoding v2.3.0
   Compiling base64 v0.13.1
   Compiling percent-encoding-rfc3986 v0.1.3
   Compiling log v0.4.19
   Compiling tinyvec v1.6.0
   Compiling form_urlencoded v1.2.0
   Compiling bitcoin_hashes v0.12.0
   Compiling unicode-normalization v0.1.22
   Compiling cc v1.0.82
   Compiling idna v0.4.0
   Compiling url v2.4.0
   Compiling secp256k1-sys v0.8.1
   Compiling secp256k1 v0.27.0
   Compiling bip21 v0.3.1
   Compiling payjoin v0.10.0 (/home/ecode/satoshi/rust-payjoin/payjoin)
    Finished dev [unoptimized + debuginfo] target(s) in 7.04s
[I] ecode@pop-os: payjoin [1.63-MSRV] » rustup show                                                                                 [15:16:39]
Default host: x86_64-unknown-linux-gnu
rustup home:  /home/ecode/.rustup

installed toolchains
--------------------

stable-x86_64-unknown-linux-gnu
nightly-x86_64-unknown-linux-gnu
1.48-x86_64-unknown-linux-gnu
1.63-x86_64-unknown-linux-gnu (default)
1.69-x86_64-unknown-linux-gnu
1.70-x86_64-unknown-linux-gnu
1.71-x86_64-unknown-linux-gnu

active toolchain
----------------

1.63-x86_64-unknown-linux-gnu (default)
rustc 1.63.0 (4b91a6ea7 2022-08-08)

[I] ecode@pop-os: payjoin [1.63-MSRV] » cargo +1.63.0 test --verbose --all-features --lib                                           [15:16:52]
error: toolchain '1.63.0-x86_64-unknown-linux-gnu' is not installed
[I] ecode@pop-os: payjoin [1.63-MSRV] » cargo +1.63 test --verbose --all-features --lib                                           1 [15:19:05]
   Compiling cfg-if v1.0.0
   Compiling serde v1.0.183
   Compiling proc-macro2 v1.0.66
   Compiling serde_derive v1.0.183
   Compiling unicode-ident v1.0.11
     Running `rustc --crate-name cfg_if --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/cfg-if-1.0.0/src/lib.rs --e
     Running `rustc --crate-name build_script_build --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/serde-1.0.183/b
     Running `rustc --crate-name build_script_build --edition=2021 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/proc-macro2-1.0
     Running `rustc --crate-name serde_derive /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/serde_derive-1.0.183/src/lib.rs --er
   Compiling untrusted v0.7.1
   Compiling spin v0.5.2
   Compiling once_cell v1.18.0
   Compiling ppv-lite86 v0.2.17
   Compiling pkg-config v0.3.27
   Compiling crc32fast v1.3.2
     Running `rustc --crate-name unicode_ident --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/unicode-ident-1.0.11
     Running `rustc --crate-name untrusted --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/untrusted-0.7.1/src/untr
     Running `rustc --crate-name spin /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/spin-0.5.2/src/lib.rs --error-format=json --
     Running `rustc --crate-name once_cell --edition=2021 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/once_cell-1.18.0/src/lib
   Compiling serde_json v1.0.104
     Running `rustc --crate-name ppv_lite86 --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/ppv-lite86-0.2.17/src/l
     Running `rustc --crate-name build_script_build /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/crc32fast-1.3.2/build.rs --err
     Running `rustc --crate-name build_script_build --edition=2021 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/serde_json-1.0.
     Running `rustc --crate-name pkg_config /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/pkg-config-0.3.27/src/lib.rs --error-f
   Compiling adler v1.0.2
     Running `rustc --crate-name adler /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/adler-1.0.2/src/lib.rs --error-format=json
   Compiling memchr v2.5.0
     Running `rustc --crate-name build_script_build --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/memchr-2.5.0/bu
   Compiling rustls v0.20.8
     Running `rustc --crate-name build_script_build --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/rustls-0.20.8/b
       Fresh base64 v0.13.1
   Compiling bitcoin v0.30.1
     Running `rustc --crate-name build_script_build --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/bitcoin-0.30.1/
   Compiling thiserror v1.0.44
     Running `rustc --crate-name build_script_build --edition=2021 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/thiserror-1.0.4
   Compiling anyhow v1.0.72
     Running `rustc --crate-name build_script_build --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/anyhow-1.0.72/b
       Fresh bech32 v0.9.1
   Compiling ryu v1.0.15
     Running `rustc --crate-name ryu --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/ryu-1.0.15/src/lib.rs --error-
   Compiling minreq v2.8.1
     Running `rustc --crate-name build_script_build --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/minreq-2.8.1/bu
   Compiling itoa v1.0.9
   Compiling rustix v0.38.7
     Running `rustc --crate-name build_script_build --edition=2021 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/rustix-0.38.7/b
     Running `rustc --crate-name itoa --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/itoa-1.0.9/src/lib.rs --error
   Compiling byteorder v1.4.3
     Running `rustc --crate-name byteorder --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/byteorder-1.4.3/src/lib.
       Fresh tinyvec_macros v0.1.1
   Compiling log v0.4.19
     Running `rustc --crate-name log /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/log-0.4.19/src/lib.rs --error-format=json --j
   Compiling linux-raw-sys v0.4.5
     Running `rustc --crate-name linux_raw_sys --edition=2021 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/linux-raw-sys-0.4.5/
   Compiling regex-syntax v0.7.4
     Running `rustc --crate-name regex_syntax --edition=2021 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/regex-syntax-0.7.4/sr
   Compiling bitflags v2.3.3
     Running `rustc --crate-name bitflags --edition=2021 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/bitflags-2.3.3/src/lib.rs
       Fresh percent-encoding v2.3.0
   Compiling fastrand v2.0.0
     Running `rustc --crate-name fastrand --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/fastrand-2.0.0/src/lib.rs
   Compiling either v1.9.0
     Running `rustc --crate-name either --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/either-1.9.0/src/lib.rs --e
       Fresh unicode-bidi v0.3.13
       Fresh percent-encoding-rfc3986 v0.1.3
   Compiling termcolor v1.2.0
     Running `rustc --crate-name termcolor --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/termcolor-1.2.0/src/lib.
   Compiling humantime v2.1.0
     Running `rustc --crate-name humantime --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/humantime-2.1.0/src/lib.
   Compiling miniz_oxide v0.7.1
     Running `rustc --crate-name miniz_oxide --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/miniz_oxide-0.7.1/src/
     Running `/home/ecode/satoshi/rust-payjoin/payjoin/target/debug/build/crc32fast-d495831a59ec6094/build-script-build`
     Running `/home/ecode/satoshi/rust-payjoin/payjoin/target/debug/build/serde-31f7a89592a43403/build-script-build`
     Running `/home/ecode/satoshi/rust-payjoin/payjoin/target/debug/build/serde_json-0b6b08b2aff440d1/build-script-build`
     Running `/home/ecode/satoshi/rust-payjoin/payjoin/target/debug/build/proc-macro2-51f81c8e084967e2/build-script-build`
       Fresh tinyvec v1.6.0
     Running `/home/ecode/satoshi/rust-payjoin/payjoin/target/debug/build/memchr-a715959dca1c5edc/build-script-build`
     Running `/home/ecode/satoshi/rust-payjoin/payjoin/target/debug/build/bitcoin-6478d6baf91332d7/build-script-build`
       Fresh form_urlencoded v1.2.0
     Running `/home/ecode/satoshi/rust-payjoin/payjoin/target/debug/build/minreq-c335c942fda5888c/build-script-build`
     Running `/home/ecode/satoshi/rust-payjoin/payjoin/target/debug/build/thiserror-b72b8b1c885623e0/build-script-build`
     Running `/home/ecode/satoshi/rust-payjoin/payjoin/target/debug/build/anyhow-9ae3d4f896804730/build-script-build`
       Fresh libc v0.2.147
       Fresh bitcoin-private v0.1.0
       Fresh hex_lit v0.1.1
     Running `rustc --crate-name crc32fast /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/crc32fast-1.3.2/src/lib.rs --error-form
       Fresh unicode-normalization v0.1.22
     Running `rustc --crate-name memchr --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/memchr-2.5.0/src/lib.rs --e
     Running `rustc --crate-name proc_macro2 --edition=2021 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/proc-macro2-1.0.66/src
     Running `rustc --crate-name serde --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/serde-1.0.183/src/lib.rs --e
       Fresh cc v1.0.82
   Compiling getrandom v0.2.10
     Running `rustc --crate-name getrandom --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/getrandom-0.2.10/src/lib
   Compiling time v0.1.45
     Running `rustc --crate-name time /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/time-0.1.45/src/lib.rs --error-format=json -
   Compiling filetime v0.2.22
     Running `rustc --crate-name filetime --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/filetime-0.2.22/src/lib.r
   Compiling xattr v1.0.1
     Running `rustc --crate-name xattr --edition=2021 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/xattr-1.0.1/src/lib.rs --err
   Compiling which v4.4.0
     Running `rustc --crate-name which --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/which-4.4.0/src/lib.rs --err
   Compiling atty v0.2.14
     Running `rustc --crate-name atty /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/atty-0.2.14/src/lib.rs --error-format=json -
   Compiling bitcoin_hashes v0.12.0
     Running `rustc --crate-name bitcoin_hashes --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/bitcoin_hashes-0.12
     Running `/home/ecode/satoshi/rust-payjoin/payjoin/target/debug/build/rustix-59ec455514539b0f/build-script-build`
       Fresh idna v0.4.0
   Compiling ring v0.16.20
     Running `rustc --crate-name build_script_build --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.16.20/bu
   Compiling bzip2-sys v0.1.11+1.0.8
     Running `rustc --crate-name build_script_build /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/bzip2-sys-0.1.11+1.0.8/build.r
     Running `rustc --crate-name anyhow --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/anyhow-1.0.72/src/lib.rs --
   Compiling rand_core v0.6.4
     Running `rustc --crate-name rand_core --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/rand_core-0.6.4/src/lib.
   Compiling flate2 v1.0.26
     Running `rustc --crate-name flate2 --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/flate2-1.0.26/src/lib.rs --
   Compiling tar v0.4.40
     Running `rustc --crate-name tar --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/tar-0.4.40/src/lib.rs --error-
       Fresh url v2.4.0
     Running `rustc --crate-name rustix --edition=2021 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/rustix-0.38.7/src/lib.rs --
   Compiling quote v1.0.32
     Running `rustc --crate-name quote --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/quote-1.0.32/src/lib.rs --er
   Compiling aho-corasick v1.0.2
     Running `rustc --crate-name aho_corasick --edition=2021 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/aho-corasick-1.0.2/sr
   Compiling rand_chacha v0.3.1
     Running `rustc --crate-name rand_chacha --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/rand_chacha-0.3.1/src/
       Fresh secp256k1-sys v0.8.1
     Running `/home/ecode/satoshi/rust-payjoin/payjoin/target/debug/build/bzip2-sys-faa844c9cba107dd/build-script-build`
   Compiling syn v2.0.28
     Running `rustc --crate-name syn --edition=2021 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/syn-2.0.28/src/lib.rs --error-
   Compiling rand v0.8.5
     Running `rustc --crate-name rand --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/rand-0.8.5/src/lib.rs --error
     Running `/home/ecode/satoshi/rust-payjoin/payjoin/target/debug/build/ring-f2692afb778ed36e/build-script-build`
     Running `rustc --crate-name bzip2_sys /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/bzip2-sys-0.1.11+1.0.8/lib.rs --error-f
   Compiling bzip2 v0.4.4
     Running `rustc --crate-name bzip2 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/bzip2-0.4.4/src/lib.rs --error-format=json
   Compiling regex-automata v0.3.6
     Running `rustc --crate-name regex_automata --edition=2021 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/regex-automata-0.3.
   Compiling tempfile v3.7.1
     Running `rustc --crate-name tempfile --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/tempfile-3.7.1/src/lib.rs
     Running `rustc --crate-name ring --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/ring-0.16.20/src/lib.rs --err
     Running `/home/ecode/satoshi/rust-payjoin/payjoin/target/debug/build/rustls-8b13f186fa4d430b/build-script-build`
     Running `rustc --crate-name bitcoin_hashes --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/bitcoin_hashes-0.12
     Running `rustc --crate-name serde_json --edition=2021 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/serde_json-1.0.104/src/
   Compiling thiserror-impl v1.0.44
     Running `rustc --crate-name thiserror_impl --edition=2021 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/thiserror-impl-1.0.
   Compiling webpki v0.22.0
   Compiling sct v0.7.0
     Running `rustc --crate-name webpki --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/webpki-0.22.0/src/lib.rs --
     Running `rustc --crate-name sct --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/sct-0.7.0/src/lib.rs --error-f
   Compiling webpki-roots v0.22.6
     Running `rustc --crate-name rustls --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/rustls-0.20.8/src/lib.rs --
     Running `rustc --crate-name webpki_roots --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/webpki-roots-0.22.6/s
   Compiling jsonrpc v0.13.0
     Running `rustc --crate-name jsonrpc --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/jsonrpc-0.13.0/src/lib.rs
   Compiling regex v1.9.3
     Running `rustc --crate-name regex --edition=2021 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/regex-1.9.3/src/lib.rs --err
   Compiling secp256k1 v0.27.0
     Running `rustc --crate-name secp256k1 --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/secp256k1-0.27.0/src/lib
   Compiling env_logger v0.9.3
     Running `rustc --crate-name env_logger --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/env_logger-0.9.3/src/li
     Running `rustc --crate-name bitcoin --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/bitcoin-0.30.1/src/lib.rs
     Running `rustc --crate-name thiserror --edition=2021 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/thiserror-1.0.44/src/lib
   Compiling zip v0.5.13
     Running `rustc --crate-name zip --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/zip-0.5.13/src/lib.rs --error-
     Running `rustc --crate-name minreq --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/minreq-2.8.1/src/lib.rs --e
   Compiling bitcoind v0.31.1
     Running `rustc --crate-name build_script_build --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/bitcoind-0.31.1
     Running `/home/ecode/satoshi/rust-payjoin/payjoin/target/debug/build/bitcoind-2ad00313e7ce2d99/build-script-build`
   Compiling core-rpc-json v0.17.0
   Compiling bip21 v0.3.1
     Running `rustc --crate-name core_rpc_json --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/core-rpc-json-0.17.0
     Running `rustc --crate-name bip21 --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/bip21-0.3.1/src/lib.rs --err
   Compiling core-rpc v0.17.0
     Running `rustc --crate-name core_rpc --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/core-rpc-0.17.0/src/lib.r
     Running `rustc --crate-name bitcoind --edition=2018 /home/ecode/.cargo/registry/src/github.com-1ecc6299db9ec823/bitcoind-0.31.1/src/lib.r
   Compiling payjoin v0.10.0 (/home/ecode/satoshi/rust-payjoin/payjoin)
     Running `rustc --crate-name payjoin --edition=2021 src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --emit=dep-info,link -C embed-bitcode=no -C debuginfo=2 --test --cfg 'feature="base64"' --cfg 'feature="rand"' --cfg 'feature="receive"' --cfg 'feature="send"' -C metadata=9b26daf276d7b76d -C extra-filename=-9b26daf276d7b76d --out-dir /home/ecode/satoshi/rust-payjoin/payjoin/target/debug/deps -C incremental=/home/ecode/satoshi/rust-payjoin/payjoin/target/debug/incremental -L dependency=/home/ecode/satoshi/rust-payjoin/payjoin/target/debug/deps --extern bip21=/home/ecode/satoshi/rust-payjoin/payjoin/target/debug/deps/libbip21-f8b47a13c0244e91.rlib --extern bitcoin=/home/ecode/satoshi/rust-payjoin/payjoin/target/debug/deps/libbitcoin-6c7cdf12a332db93.rlib --extern bitcoind=/home/ecode/satoshi/rust-payjoin/payjoin/target/debug/deps/libbitcoind-8e902a353562bb86.rlib --extern env_logger=/home/ecode/satoshi/rust-payjoin/payjoin/target/debug/deps/libenv_logger-4b150360b2b3b536.rlib --extern log=/home/ecode/satoshi/rust-payjoin/payjoin/target/debug/deps/liblog-242a499eff9d6bbf.rlib --extern rand=/home/ecode/satoshi/rust-payjoin/payjoin/target/debug/deps/librand-3267c758f50ab24c.rlib --extern url=/home/ecode/satoshi/rust-payjoin/payjoin/target/debug/deps/liburl-1e178e0284905a4a.rlib -L native=/home/ecode/satoshi/rust-payjoin/payjoin/target/debug/build/secp256k1-sys-23189fba074aca69/out`
    Finished test [unoptimized + debuginfo] target(s) in 18.29s
     Running `/home/ecode/satoshi/rust-payjoin/payjoin/target/debug/deps/payjoin-9b26daf276d7b76d`

running 17 tests
test input_type::tests::test_p2pk ... ok
test input_type::tests::test_p2pkh ... ok
test input_type::tests::test_p2sh ... ok
test input_type::tests::test_p2wpkh ... ok
test input_type::tests::test_p2sh_p2wpkh ... ok
test input_type::tests::test_p2wsh ... ok
test input_type::tests::test_p2sh_p2wsh ... ok
test uri::tests::test_todo_url_encoded ... ignored
test uri::tests::test_short ... ok
test receive::test::can_get_proposal_from_request ... ok
test receive::test::unchecked_proposal_unlocks_after_checks ... ok
test uri::tests::test_missing_amount ... ok
test uri::tests::test_unsupported ... ok
test uri::tests::test_unencrypted ... ok
test uri::tests::test_valid_url ... ok
test send::tests::official_vectors ... ok
test uri::tests::test_valid_uris ... ok

test result: ok. 16 passed; 0 failed; 1 ignored; 0 measured; 0 filtered out; finished in 0.00s
```